pylabrobot/visualizer/lib.js

```diff
@@ -825,10 +943,41 @@ class Trash extends Resource {
   }
 }
 
-// Nothing special.
-class Carrier extends Resource {}
-class PlateCarrier extends Carrier {}
-class TipCarrier extends Carrier {}
+class Carrier extends Resource {
+  getColor() {
+    return RESOURCE_COLORS["Carrier"];
+  }
+}
+
+class MFXCarrier extends Carrier {
+  getColor() {
+    return RESOURCE_COLORS["MFXCarrier"];
+  }
+}
+
+class PlateCarrier extends Carrier {
+  getColor() {
+    return RESOURCE_COLORS["PlateCarrier"];
+  }
+}
+
+class TipCarrier extends Carrier {
+  getColor() {
+    return RESOURCE_COLORS["TipCarrier"];
+  }
+}
+
+class TroughCarrier extends Carrier {
+  getColor() {
+    return RESOURCE_COLORS["TroughCarrier"];
+  }
+}
+
+class TubeCarrier extends Carrier {
+  getColor() {
+    return RESOURCE_COLORS["TubeCarrier"];
+  }
+}
```

Comment:
Considering Resource dynamically loads the color based on the constructor name, I think these are not needed.